contrib/mesos/pkg/minion/server.go

```diff
@@ -258,15 +215,54 @@ func (ms *MinionServer) Run(hks hyperkube.Interface, _ []string) error {
 	ms.clientConfig = clientConfig
 
 	// run subprocesses until ms.done is closed on return of this function
-	defer close(ms.done)
 	if ms.runProxy {
-		go runtime.Until(ms.launchProxyServer, 5*time.Second, ms.done)
+		ms.launchProxyServer()
 	}
-	go runtime.Until(ms.launchExecutorServer, 5*time.Second, ms.done)
 
-	// wait until minion exit is requested
-	// don't close ms.exit here to avoid panics of go routines writing an error to it
-	return <-ms.exit
+	// abort closes when the kubelet-executor dies
+	abort := ms.launchExecutorServer()
+	shouldQuit := termSignalListener(abort)
+	te := mergeTaskOutput(ms.tasks, shouldQuit)
+
+	// TODO(jdef) do something fun here, such as reporting task completion to the apiserver
```

Comment:
I am a bit puzzled why we need this error and task completion aggregation logic in the taskEvents interface. Here, right now we don't need it. Do we really need it for some later apiserver task completion notification? In the current state it feels very over-engineered.